src/smart-components/order-process/order-processes.js

```diff
@@ -0,0 +1,190 @@
+    return dispatch(fetchOrderProcesses(meta)).then(() =>
+      filteringCallback(false)
+    );
+  },
+  1000
+);
+
+const prepareChips = (filterValue, intl) =>
+  filterValue
+    ? [
+        {
+          category: intl.formatMessage(tableToolbarMessages.name),
+          key: 'name',
+          chips: [{ name: filterValue, value: filterValue }]
+        }
+      ]
+    : [];
+
+const initialState = (filterValue = '') => ({
+  filterValue,
+  isOpen: false,
+  isFetching: true,
+  isFiltering: false
+});
+
+const orderProcessesState = (state, action) => {
+  switch (action.type) {
+    case 'setFetching':
+      return { ...state, isFetching: action.payload };
+    case 'setFilterValue':
+      return { ...state, filterValue: action.payload };
+    case 'setFilteringFlag':
+      return { ...state, isFiltering: action.payload };
+    default:
+      return state;
+  }
+};
+
+const OrderProcesses = () => {
+  const {
+    orderProcesses: { data, meta },
+    sortBy,
+    filterValueRedux
+  } = useSelector(
+    ({
+      orderProcessReducer: {
+        orderProcesses,
+        sortBy,
+        filterValue: filterValueRedux
+      }
+    }) => ({ orderProcesses, sortBy, filterValueRedux }),
+    shallowEqual
+  );
+  const [{ filterValue, isFetching, isFiltering }, stateDispatch] = useReducer(
+    orderProcessesState,
+    initialState(filterValueRedux)
+  );
+
+  const dispatch = useDispatch();
+  const intl = useIntl();
+
+  const updateOrderProcesses = (pagination) => {
+    stateDispatch({ type: 'setFetching', payload: true });
+    return dispatch(fetchOrderProcesses(pagination))
+      .then(() => stateDispatch({ type: 'setFetching', payload: false }))
+      .catch(() => stateDispatch({ type: 'setFetching', payload: false }));
+  };
+
+  useEffect(() => {
+    updateOrderProcesses(defaultSettings);
+    scrollToTop();
+  }, []);
+
+  const handleFilterChange = (value) => {
+    stateDispatch({ type: 'setFilterValue', payload: value });
+    debouncedFilter(
+      value,
+      dispatch,
+      (isFiltering) =>
+        stateDispatch({ type: 'setFilteringFlag', payload: isFiltering }),
+      { ...meta, offset: 0 }
+    );
+  };
+
+  const onSort = (_e, index, direction, { property }) => {
+    dispatch(sortOrderProcesses({ index, direction, property }));
+    return updateOrderProcesses();
+  };
+
+  return (
+    <Fragment>
+      <TopToolbar>
+        <TextContent className="pf-u-ml-md">
+          <Text component={TextVariants.h1}>
+            {intl.formatMessage(orderProcessesMessages.title)}
+          </Text>
+        </TextContent>
+      </TopToolbar>
+      <TableToolbarView
+        sortBy={sortBy}
+        onSort={onSort}
+        data={data}
+        createRows={createRows}
+        columns={columns(intl)}
+        fetchData={updateOrderProcesses}
+        titlePlural={intl.formatMessage(orderProcessesMessages.orderProcesses)}
+        titleSingular={intl.formatMessage(orderProcessesMessages.orderProcess)}
```

Comment:
This should be handled by the react int and some variable.

Reply:
the singular and plural are used in different places.